internal/controller/bucket_controller.go

```diff
@@ -590,6 +607,12 @@ func (r *BucketReconciler) reconcileDelete(ctx context.Context, obj *sourcev1.Bu
 	// Remove our finalizer from the list
 	controllerutil.RemoveFinalizer(obj, sourcev1.SourceFinalizer)
 
+	// Cleanup caches.
+	if r.TokenCache != nil {
```

Comment:
There's already a guard inside `.DeleteEventsForObject()` 👍 We can remove this `if` like in all other controllers